File: src/ts/radio/apiDebug.ts
import {ApiModule, SendResult} from './apiModule';

export class DebugApi extends ApiModule {
    /**
     * Ping 远端，等待 pong 回复。
     */
    ping(): SendResult<any> {
        return this.send('ping', null, 'pong');
    }
}

File: src/rs/radio/api_debug.rs
use crate::radio::api_module::{ApiModule, SendResult};

pub struct DebugApi {
    pub base: ApiModule,
}

impl DebugApi {
    pub fn new(base: ApiModule) -> Self {
        Self { base }
    }

    pub fn ping(&self) -> SendResult {
        self.base.send("ping", None, Some("pong"))
    }
}

impl Default for DebugApi {
    fn default() -> Self {
        Self { base: ApiModule::default() }
    }
}
